src/pages/transfer/Transfer.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useState } from 'react';
+import { useSearchParams, useNavigate } from 'react-router-dom';
+import {
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  TextField,
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+  Container,
+  Box,
+  CircularProgress,
+} from '@mui/material';
+import styles from './Transfer.module.css';
+import { postTransaction } from '../../api/account/AccountApi';
+import { useAppSelector } from '../../redux/hooks';
+import getExchangeRates from '../../utils/utils';
+import { CURRENCY_OPTIONS } from '../../constants/currency';
+
+function Transfer() {
+  const navigate = useNavigate();
+  const accounts = useAppSelector((state) => state.account.accounts);
+  const [searchParams] = useSearchParams();
+
+  const [account, setAccount] = useState<number>(
+    Number(searchParams.get('id')),
+  );
+  const [currency, setCurrency] = useState(searchParams.get('currency' || ''));
+  const [amount, setAmount] = useState<number>();
+  const [destination, setDestination] = useState<number>();
+  const [reference, setReference] = useState('');
+  const [confirmOpen, setConfirmOpen] = useState(false);
+  const [error, setError] = useState('');
+  const [errorOpen, setErrorOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [originAmount, setOriginAmount] = useState(0);
+  const [originCurrency, setOriginCurrency] = useState('');
+
+  const handleConfirm = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const currentAccount = accounts.filter((a) => a.id === account && a);
```

Comment:
```suggestion
    const currentAccount = accounts.filter((account) => account && account.id === accountId);
```

Es confuso tener un id guardado bajo el nombre de una entidad completa.
Si tenés una variable con el ID de una account, va de nombre `accountId` para evitar malentendidos.

Reply:
Otra cosa, porque se chequea si la `account` está definida primero?
Puede venir en `null` o `undefined`?